src/app/src/androidTest/java/com/example/bookmark/SignupActivityTest.java

```diff
@@ -0,0 +1,68 @@
+package com.example.bookmark;
+
+import android.app.Activity;
+import android.view.View;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.rule.ActivityTestRule;
+
+import com.robotium.solo.Solo;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+
+/**
+ * Perform intent testing on the SignupActivity
+ * <p>
+ * Outstanding Issues/TODOs
+ * Test Signing Up (how to make sure that user exists?)
+ *
+ * @author Konrad Staniszewski.
+ */
+public class SignupActivityTest {
+    private Solo solo;
+    @Rule
+    public ActivityTestRule<SignupActivity> rule =
+        new ActivityTestRule<>(SignupActivity.class, true, true);
```

Comment:
Within this pull request, you vary your capitalization of sign up. Here it's `Signup` as if "signup" is one word, but in other places it's "sign up" (two words). I would consistently use `signUp` or `SignUp` throughout your code since "signup" is not a word as far as I'm aware.

This is rather minor, but it's easier to fix now than try to address after more classes refer to SignupActivity.